src/field/vecops.cxx

```diff
@@ -396,250 +396,216 @@ const Field3D V_dot_Grad(const Vector3D &v, const Field3D &f) {
   return result;
 }
 
-const Vector2D V_dot_Grad(const Vector2D &v, const Vector2D &a, CELL_LOC outloc) {
+const Vector2D V_dot_Grad(const Vector2D &v, const Vector2D &a) {
   TRACE("V_dot_Grad( Vector2D , Vector2D )");
 
-  if (outloc == CELL_DEFAULT) {
-    outloc = v.getLocation();
-  }
-
-  ASSERT1(outloc != CELL_VSHIFT);
   ASSERT1(v.getLocation() == a.getLocation());
-  // Note the following assert means the outloc argument is pointless
-  ASSERT1(v.getLocation() == outloc);
 
-  Mesh *localmesh = v.x.getMesh();
-  Vector2D result(localmesh);
+  Vector2D result{v.x.getMesh()};
 
-  Coordinates *metric = localmesh->coordinates(outloc);
+  auto metric_x = v.x.getCoordinates();
+  auto metric_y = v.y.getCoordinates();
```

Comment:
These all have to be at exactly the same location so we only really need one of the three.

Reply:
Ah, I didn't catch that it doesn't support `VSHIFT` for the input locations